ngraph/core/reference/src/runtime/opt_kernel/parallel_executor.cpp

```diff
@@ -0,0 +1,58 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/opt_kernel/parallel_executor.hpp"
+#include "ngraph/env_util.hpp"
+
+using namespace ngraph;
+
+bool runtime::detail::forced_single_threaded_execution()
+{
+    return getenv_bool("REF_SINGLE_THREADED", false);
+}
+
+uint64_t runtime::detail::parallelism_threshold()
+{
+    const int32_t DEFAULT_THRESHOLD = 1000000;
+    const auto t = getenv_int("REF_THRESHOLD", DEFAULT_THRESHOLD);
+    if (t < 0)
+    {
+        return DEFAULT_THRESHOLD;
+    }
```

Comment:
do we need this code here?

Reply:
Probably it's a quickest way to validate `t` value since `getenv_int` can return negative value with underlaying `strtol` function. For safety it should stay or be refined into something better.

Reply:
Yes, this check is intentional to make sure that the user doesn't do something weird by specifying negative values. The same goes for the other env variable (tasks number).

Reply:
in this context(Threshold) negative value is nothing bad IMO it just turn off the threshold.

Reply:
But switching the threshold off causes performance drop for small tensors. Perhaps the default value should be different (like a 100K for example) but I don't see any good reason to allow negative values.